### PR TITLE
Allow state method to receive env

### PR DESCRIPTION
**Pull request by eszjay**
I have a state algorithm that requires access to values in env. This change allows it to receive that reference in a backwards compatible way and brings the state method in line with the setup method.

### CODE REVIEW

File: lib/omniauth/strategies/openid_connect.rb [hunk 1]
```diff
@@ -203,7 +203,9 @@ def client_options
       end
 
       def new_state
-        state = options.state.call if options.state.respond_to? :call
+        if options.state.respond_to? :call
+          state = options.state.arity == 1 ? options.state.call(env) : options.state.call
```
Comment (m0n9oose) [inline lib/omniauth/strategies/openid_connect.rb]: 1. shouldn't you check arity of `call` with `options.state.method(:call).arity == 1` ?
2. please consider change `state` variable definition to something like this:

```ruby
state = if options.state.respond_to?(:call)
          if options.state.method(:call).arity == 1
            options.state.call(env)
          else
            options.state.call
          end
        end
```

It's more verbose but way more readable.

[the remark this answers is not in the document]
Comment (eszjay) [reply lib/omniauth/strategies/openid_connect.rb]: I see that the incoming rubocop standard is 130 columns per line. Is the 80 character limit real?

Comment (eszjay) [reply lib/omniauth/strategies/openid_connect.rb]: It won't work quite like that, but we could eliminate the ternary operator.

options.state is a method, so it does not exactly have methods.
You can see the arity usage examples at https://ruby-doc.org/core-2.6.1/Method.html#method-i-arity

The ternary operator is suggested by rubocop (search for ternary on https://github.com/rubocop-hq/ruby-style-guide ) but does not make much difference here and your suggestions is clean.

The precise formatting is your call, as it is your project. Uniformity and maintainability are significant concerns for a project.

[the remark this answers is not in the document]
Comment (m0n9oose) [reply lib/omniauth/strategies/openid_connect.rb]: @eszjay I'm gonna add custom rubocop config with increased to 130 chars value soon

**Comment by m0n9oose** [comment]
Did you run the tests? I'd also kindly ask you to add some test suits to cover the new code.

**Comment by m0n9oose** [comment]
@eszjay please add test suits to cover the changed code

**Comment by m0n9oose** [comment]
@eszjay please cover the changes with a couple of test suites. Otherwise, I'll have to close the PR.

File: test/lib/omniauth/strategies/openid_connect_test.rb [hunk 2]
```diff
@@ -330,14 +330,14 @@ def test_failure_endpoint_redirect
 
       def test_state
         strategy.options.state = lambda { 42 }
-        session = { "state" => 42 }
 
-        expected_redirect = /&state=/
+        expected_redirect = /&state=42/
```
Comment (eszjay) [inline test/lib/omniauth/strategies/openid_connect_test.rb]: The existing state test was not verifying the value from the state function

**Pull request closed without merging**